### PR TITLE
utter_message explicitly takes in response and template

### PR DESCRIPTION
**Pull request by yennycheung**
**Proposed changes**:
- `utter_message` explicitly takes in `response` and `template`

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa-sdk/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

**Comment by yennycheung** [comment]
Publishing the branch, but I'm pretty sure there are some tests to fix / add. For some reason I couldn't run `make test` in local.

File: rasa_sdk/executor.py [hunk 1]
```diff
@@ -33,19 +33,27 @@ def utter_message(
...
+        response: Optional[Text] = None,
         attachment: Optional[Text] = None,
         buttons: Optional[List[Dict[Text, Any]]] = None,
         elements: Optional[List[Dict[Text, Any]]] = None,
         **kwargs: Any,
     ) -> None:
-        """"Send a text to the output channel"""
-
+        """Send a text to the output channel."""
+        if template and not response:
```
Comment (wochinge) [inline rasa_sdk/executor.py:43]: why not this?
```suggestion
        if template:
```

File: rasa_sdk/executor.py [hunk 2]
```diff
@@ -33,19 +33,27 @@ def utter_message(
...
+            response = template
+            warnings.warn(
+                "Please pass the parameter `response` instead of `template` "
+                "to utter_message. `template` will be deprecated in Rasa 3.0.0. ",
```
Comment (wochinge) [inline rasa_sdk/executor.py:47]:
```suggestion
                "to 'utter_message'. `template` will be deprecated in Rasa 3.0.0. ",
```

**Review by wochinge — APPROVED** [review]
Slick 💯

File: rasa_sdk/executor.py (hunk 1 above)
Comment (yennycheung) [reply rasa_sdk/executor.py:43]: I was thinking `response` should be the source of truth. If `template` was used, it still shouldn't overwrite `response` if it wasn't not empty. 🤷🏻‍♀️

Comment (wochinge) [reply rasa_sdk/executor.py:43]: sounds good

**Review by wochinge — APPROVED** [review]
Ready to merge!

**Pull request merged**